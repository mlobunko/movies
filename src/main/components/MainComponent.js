import React from 'react';

import Header from 'app/header/components';
import Backdrop from 'app/backdrop/components';
import TrailerOrPoster from 'app/trailerOrPoster/components';
import { StyledMainComponent } from './styles';

export const MainComponent = () => (
  <StyledMainComponent>
    <Header />
    <Backdrop>
      <TrailerOrPoster />
    </Backdrop>
  </StyledMainComponent>
);

MainComponent.displayName = 'MainComponent';

export default MainComponent;
