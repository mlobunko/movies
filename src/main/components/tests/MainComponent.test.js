import React from 'react';
import { shallow } from 'enzyme';

import { MainComponent } from '../MainComponent';

describe('render', () => {
  it('renders correctly, error true', () => {
    const wrapper = shallow(<MainComponent error={'error'} />);
    expect(wrapper).toMatchSnapshot();
  });

  it('renders correctly, error false', () => {
    const wrapper = shallow(<MainComponent />);
    expect(wrapper).toMatchSnapshot();
  });
});
