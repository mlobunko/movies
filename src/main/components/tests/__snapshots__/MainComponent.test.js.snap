// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render renders correctly, error false 1`] = `
<StyledMainComponent>
  <ErrorBoundary>
    <Header />
    <Connect(BackdropComponent)>
      <Connect(SearchInputContainer) />
      <StyledContent>
        <Connect(PosterComponent) />
        <Connect(DescriptionComponent) />
      </StyledContent>
    </Connect(BackdropComponent)>
  </ErrorBoundary>
</StyledMainComponent>
`;

exports[`render renders correctly, error true 1`] = `
<StyledMainComponent>
  <ErrorBoundary>
    <Header />
    <h1
      style={
        Object {
          "textAlign": "center",
        }
      }
    >
      Something went wrong.
    </h1>
  </ErrorBoundary>
</StyledMainComponent>
`;
